src/components/filterable-list/FilterableList.js

```diff
@@ -61,6 +62,12 @@ export default class FilterableList extends PureComponent {
     })
   }
 
+  shouldComponentUpdate(nextProps) {
+    if (nextProps.isFilterFocused !== this.props.isFilterFocused) return true
```

Comment:
How does this solve problem? if the list updates it would still render a new one. How about ordering the names?

Also can you add an explanation and linking to the issue as comment. This is important to not forget why this is needed.